packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -281,247 +285,85 @@ function contextualizeInsertedArrayContent(
-	Array.prototype.map,
-	// Array.prototype.pop,
-	// Array.prototype.push,
-	Array.prototype.reduce,
-	Array.prototype.reduceRight,
-	// Array.prototype.reverse,
-	// Array.prototype.shift,
-	Array.prototype.slice,
-	Array.prototype.some,
-	// Array.prototype.sort,
-	// Array.prototype.splice,
-	Array.prototype.toLocaleString,
-	Array.prototype.toString,
-	// Array.prototype.unshift,
-	Array.prototype.values,
-].forEach((fn) => {
-	arrayNodePrototypeProperties[fn.name] = { value: fn };
+const arrayPrototypeKeys = [
+	"concat",
+	"entries",
+	"every",
+	"filter",
+	"find",
+	"findIndex",
+	"flat",
+	"flatMap",
+	"forEach",
+	"includes",
+	"indexOf",
+	"join",
+	"keys",
+	"lastIndexOf",
+	"map",
+	"reduce",
+	"reduceRight",
+	"slice",
+	"some",
+	"toLocaleString",
+	"toString",
+	"values",
+
+	// "copyWithin",
+	// "fill",
+	// "length",
+	// "pop",
+	// "push",
+	// "reverse",
+	// "shift",
+	// "sort",
+	// "splice",
+	// "unshift",
+] as const;
+
+/**
+ * {@link TreeNodeValid}, but modified to add members from Array.prototype named in {@link arrayPrototypeKeys}.
+ */
+abstract class TreeNodeWithArrayFeaturesUntyped<
+	const T extends ImplicitAllowedTypes,
+> extends TreeNodeValid<Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>> {}
+
+arrayPrototypeKeys.forEach((key) => {
+	Object.defineProperty(TreeNodeWithArrayFeaturesUntyped.prototype, key, {
+		value: Array.prototype[key],
+	});
 });
 
-/* eslint-enable @typescript-eslint/unbound-method */
+/**
+ * Type of {@link TreeNodeWithArrayFeaturesUntyped}, but with its array members added to the type.
+ */
+type TreeNodeWithArrayFeaturesType = TreeNodeValueStatics &
+	(abstract new <T extends ImplicitAllowedTypes>(
+		input: Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>> | InternalTreeNode,
+	) => TreeNodeValid<Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>> &
+		Pick<readonly TreeNodeFromImplicitAllowedTypes<T>[], (typeof arrayPrototypeKeys)[number]>);
 
-// #endregion
+/**
+ * {@link TreeNodeValid}, but modified to add members from Array.prototype named in {@link arrayPrototypeKeys}.
+ */
+const TreeNodeWithArrayFeatures =
+	TreeNodeWithArrayFeaturesUntyped as unknown as TreeNodeWithArrayFeaturesType;
+
+type NonConstructorKeys<T> = { [P in keyof T]: T[P] extends new () => any ? never : P }[keyof T];
+type NonConstructor<T> = Pick<T, NonConstructorKeys<T>>;
+
+// Due to https://github.com/microsoft/TypeScript/issues/35416 this fails to preserve protected members, which are what we are trying to preserve here.
+// This results in the overridden protected members of schema in arraySchema not being able to be marked with override or checked for correct types against the inherited members.
+// TODO:
+// Implementing custom versions of all the array methods instead of reusing the versions from array would allow removing this logic, restoring safer checking of these overridden protected static methods.
+type TreeNodeValueStatics = NonConstructor<typeof TreeNodeValid>;
```

Comment:
This attempts to preserve static members of TreeNodeValid. As noted, it doesn't work for protected members. This means this change losses a bit of type safty here.